hal/inc/ble_hal.h

```diff
@@ -373,6 +373,26 @@ int hal_ble_lock(void* reserved);
  */
 int hal_ble_unlock(void* reserved);
 
+/**
+ * Notify that device has entered the Listening mode, in which case the BLE
+ * advertising data/parameters and connection parameters are not allowed to be modified.
+ *
+ * @param      reserved  The reserved
+ *
+ * @returns    0 on success, system_error_t on error.
+ */
+int hal_ble_notify_enter_listening_mode(void* reserved);
```

Comment:
Let's perhaps rename this to something more generic. HAL ideally shouldn't be aware of high-level concepts like listening mode. Something generic like `hal_ble_enter_locked_mode()` (not ideal, but naming things is difficult) sounds better IMO.